src/core/license.rs

```diff
@@ -523,15 +578,29 @@ impl ServerLicenseRequest {
                 "wBlobLen" => DynOption::new(U16::LE(0), | size | MessageOption::Size("blobData".to_string(), size.inner() as usize)),
                 "blobData" => Vec::<u8>::new()
             ],
-            "ScopeCount" => DynOption::new(U32::LE(0), | size | MessageOption::Size("ScopeArray".to_string(), size.inner() as usize)),
-            "ScopeArray" => Vec::<u8>::new()
+            "ScopeCount" => U32::LE(0)
         ];
 
         message.read(raw)?;
         let server_random = cast!(DataType::Slice, message["ServerRandom"])?;
         let version = cast!(DataType::U32, message["dwVersion"])?;
         let server_certificate = cast!(DataType::Component, message["ServerCertificate"])?;
         let mut blob_data = cast!(DataType::Slice, server_certificate["blobData"])?;
+        let scope_count = cast!(DataType::U32, message["ScopeCount"])?;
+
+        let mut scopes = Vec::with_capacity(scope_count as usize);
```

Comment:
Where is that `ScopeCount` coming from? Is it trustworthy?

Reply:
It's coming from the Windows server that is requesting a license from us.

I've never seen anything other than `1`, but I don't have a very sophisticated setup. Maybe we should put some reasonable maximum (100?) here to avoid unbounded allocations?

(Now that you point this out, the `with_capacity` doesn't do much because thats the number of scopes, not any sort of byte-length).